app/controllers/plan_changes_controller.rb

```diff
@@ -33,15 +34,11 @@ def create
 
   private
 
-  def not_authorized_user
-    head :not_authorized unless current_user
-  end
-
   def set_current_user_plan
-    @current_plan = current_user&.active_subscription
+    @current_plan = current_user&.active_subscription&.plan
   end
 
   def plan_change_params
-    params.require(:plan_change).permit(:user_id, :old_plan_id, :new_plan_id)
```

Comment:
How is that the `plan_change_params` end point doesn't need user id? It is this intended?

Reply:
No endpoint should require a user_id, this is a security issue. This means you could send a user_id and it will run these commands on behalf of other user. Instead we are taking the user_id from the current_user